scripts/openmrs/bundled-docker/update-docker-compose-env-defaults.groovy

```diff
@@ -195,27 +195,28 @@ static def processEnvVarLine(String line, Map<String, String> envDefaults) {
     def varName = varMatch[0][1]
     def currentValue = varMatch[0][2]
 
-    if (needsDefault(currentValue) && envDefaults.containsKey(varName)) {
-        def defaultValue = envDefaults[varName]
-        def newValue = "\${${varName}:-${escapeValue(defaultValue)}}"
-        def newLine = "${indent}${varName}: ${newValue}"
-
-        //log("Setting default for ${varName}")
-        return [line: newLine, changed: true]
+    def changed = false
+    def newValue = currentValue.replaceAll(/\$\{([A-Za-z_][A-Za-z0-9_]*)\}/) { full, v ->
+        if (envDefaults.containsKey(v) && !full.contains(':-')) {
+            changed = true
+            "\${${v}:-${escapeValue(envDefaults[v])}}"
+        } else {
+            full
+        }
```

Comment:
Does it support env variables like `${OPENMRS_DB_USER:-openmrs}`
Can you add a comment of what is expected at each step?